### PR TITLE
Ensure correct endian in numba_histogram

### LINKED ISSUE #2668: Histogram error with `>u2` dtype
The histogram function when plotting does not work with certain dtypes. Example of this:

```python
import dask.array as da
import hyperspy.api as hs
data = da.arange(0, 10000, dtype='>u2').reshape((100, 100))
s = hs.signals.Signal2D(data).as_lazy()
s.plot()
# Press H
```

Gives the error message:
```python
File "hyperspy/hyperspy/signal_tools.py", line 964, in plot_histogram
    self.hist_data = self._get_histogram(data)
File "hyperspy/hyperspy/signal_tools.py", line 922, in _get_histogram
    return numba_histogram(data, bins=self.bins,

File "hyperspy/hyperspy/misc/array_tools.py", line 396:
def numba_histogram(data, bins, ranges):
    <source elided>
    # Adapted from https://iscinumpy.gitlab.io/post/histogram-speeds-in-python/
    hist = np.zeros((bins,), dtype=np.intp)
    ^

This error may have been caused by the following argument(s):
- argument 0: Unsupported array dtype: >u2
```

The `>u2` dtype sometimes comes from reading raw binary files from memory, via `numpy.memmap`. Ergo, the dtype is set directly when loading the file. While it is possible to change the dtype manually to something else, like `np.uint16` it would be nice to avoid this if possible.

The error seems to stem from the `numba_histogram` function utilizing numba, which does not support these types of dtypes (like `>u2`).

I'm not sure how best to fix this. One way could be to check if the data is this type of dtype, and if it is, convert it to `np.uint16`.

### PR DESCRIPTION
**Pull request by jlaehne**
### Description of the change
Makes sure that `numba_histogram` runs for big endian datatypes like `>u2` or `>f4` by converting data to little endian for execution.

Closes #2668

### Progress of the PR
- [X] Change implemented,
- [X] extend to other numba functions,
- [x] add tests,
- [X] ready for review.

### Minimal example of the bug fix or the new feature
```python
import numpy as np
from hyperspy.misc.array_tools import numba_histogram

# works
arr = np.arange(100, dtype='<u2')
numba_histogram(arr, 5, (0, 100))

# works
arr = np.arange(100, dtype='u2')
numba_histogram(arr, 5, (0, 100))

# works
arr = np.arange(100, dtype='>u2')
numba_histogram(arr, 5, (0, 100))
```

### CODE REVIEW

**Comment by jlaehne** [comment]
Question is, whether this affects any other functions that run numba? Namely

- `_linear_bin_loop` in `array_tools.py`
- `lowess` in `lowess_smooth.py`
- `_fast_mean` and `_fast_std` in `peakfinders2D.py`

Probably, the would need a similar wrapper.

File: hyperspy/misc/array_tools.py [hunk 1]
```diff
@@ -392,6 +391,17 @@ def numba_histogram(data, bins, ranges):
     hist : array
         The values of the histogram.
     """
+    if data.dtype.byteorder == '>':
```
Comment (francisco-dlp) [inline hyperspy/misc/array_tools.py]: What about the following instead?

```python
    if data.dtype.byteorder == '>':
       data = data.byteswap().newbyteorder()
    return _numba_histogram(data, bins, ranges)
```
It is the same, but easier to maintain...

Comment (jlaehne) [reply hyperspy/misc/array_tools.py]: Sure, no problem. 

Should I add it for the other 4 numba-functions? I think we cannot rule out that someone calls them with the wrong endian. For the private functions I could just add it to the calling functions, only `lowess` would need a wrapper.

Comment (ericpre) [reply hyperspy/misc/array_tools.py]: The issue is that numba only support native endianess and this approach will not work on all platform. What's about:
```python
        # Numba only supported native dtype
        # https://github.com/numba/numba/issues/2243
        if not data.dtype.isnative:
            data = data.astype(data.dtype.type
```

Comment (jlaehne) [reply hyperspy/misc/array_tools.py]: Yes, that should do the job, indeed.

**Review by ericpre — APPROVED** [review]
Nice to take the opportunity to increase the coverage, thanks!

**Comment by jlaehne** [comment]
> Nice to take the opportunity to increase the coverage, thanks!

Not much, but at least a few obvious ones.

In the end, I refrained from covering `_fast_mean` and `_fast_std` in `peakfinders2D.py`, because I could not trigger the error using the 'stat' peakfinder routine.

**Pull request merged**